### PR TITLE
Iox #679 create introductory example of the waitset

### LINKED ISSUE #679: Create introductory example of the waitset
Create and describe the basic case of one subscriber waiting for data with the waitset (all existing examples are more advanced). This should serve as "hello world" example of the waitset.

### PR DESCRIPTION
**Pull request by MatthiasKillat**
## Pre-Review Checklist for the PR Author

1. [x] Code follows the coding style of [CONTRIBUTING.md][contributing]
1. [x] Tests follow the [best practice for testing][testing]
1. [x] Branch follows the naming format (`iox-#123-this-is-a-branch`)
1. [x] Commits messages are according to this [guideline][commit-guidelines]
    - [x] Commit messages have the issue ID (`iox-#123 commit text`)
    - [x] Commit messages are signed (`git commit -s`)
    - [x] Commit author matches [Eclipse Contributor Agreement][eca] (and ECA is signed)
1. [x] Update the PR title
   - Follow the same conventions as for commit messages
   - Link to the relevant issue
1. [x] Relevant issues are linked
1. [x] Add sensible notes for the reviewer
1. [x] All checks have passed (except `task-list-completed`)
1. [x] Assign PR to reviewer

[commit-guidelines]: https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html
[eca]: http://www.eclipse.org/legal/ECA.php
[contributing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/CONTRIBUTING.md#coding-style
[testing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/website/advanced/best-practice-for-testing.md

## Checklist for the PR Reviewer

- [x] Commits are properly organized and messages are according to the guideline
- [x] Code according to our coding style and naming conventions
- [x] Unit tests have been written for new behavior
- [x] Public API changes are documented via doxygen
- [x] Copyright owner are updated in the changed files
- [x] PR title describes the changes

## Post-review Checklist for the PR Author

1. [x] All open points are addressed and tracked via issues

## References

- Closes #679

### CODE REVIEW

File: iceoryx_examples/waitset/README.md [hunk 1]
```diff
@@ -124,21 +124,23 @@ are stored inside of the **EventInfo** and can be acquired by the user.
 
 This example consists of 5 use cases.
```
Comment (FerdinandSpitzschnueffler) [inline iceoryx_examples/waitset/README.md:125]: 6 use cases

Comment (elBoberido) [reply iceoryx_examples/waitset/README.md:125]: it seems this is the result of an party unsuccessful merge/rebase. Could maybe be fixed by some rebase black magic

Comment (MatthiasKillat) [reply iceoryx_examples/waitset/README.md:125]: It is, and it is quite likely that we will accidentally desync even more with merges and rebases when working on two (slightly) different branches.

Comment (MatthiasKillat) [reply iceoryx_examples/waitset/README.md]: Corrected by changing the last commit.

**Review by FerdinandSpitzschnueffler — APPROVED** [review]

**Review by dkroenke — APPROVED** [review]

**Pull request merged**